### ISSUE TITLE
PShape.scale not working with PShape.resetMatrix when P2D renderer is used

### ISSUE THREAD

**Issue by Reantric** [benfry/processing4#217]
## Description

PShape.scale does not play well with PShape.resetMatrix when P2D renderer is selected.
## Expected Behavior

 I believe this to be a bug. I expected the ellipse to be drawn and scaled up absolutely by 2. I tried other svg's (loadShape) and the problem still occurred.
 Without P2D: 
![image](https://user-images.githubusercontent.com/44213272/124039952-eb0fb700-d9d1-11eb-8d90-fa04269225d2.png)
With P2D:
![image](https://user-images.githubusercontent.com/44213272/124039981-febb1d80-d9d1-11eb-8573-0770b7182dfa.png)

## Current Behavior
 
The ellipse is not drawn at all. I think the issue lies within PShape.scale because PShape.translate seems to work fine, or possibly P2D and PShapeOpenGL.

## Steps to Reproduce

```processing
PShape s;

void setup(){
  size(1000,1000,P2D); // Removing P2D allows it to work.
    s = createShape(ELLIPSE,0,0,100,100);
}


void draw(){
  background(0);
  translate(width/2,height/2);
  // s.translate(-200,100); This works!
  s.scale(2);
  shape(s,0,0);
  s.resetMatrix();
} 
```

## Your Environment

 

* Processing version: 4.0a5
* Operating System and OS version: Windows 10 Build 19041.1052
* Other information:

**Comment by sampottinger** [comment]
Hey there! Just a quick note that I believe I am seeing the same behavior on Processing 3. @protonlaser91, can you please confirm if you see this in Processing 3 as well?

**Comment by Reantric** [comment]
@sampottinger Apologies for the late reply, but yes I do see this on Processing 3 too.

**Comment by codeanticode** [comment]
The ultimate cause for this issue was an error in an internal utility to invert the scaling transformation in 2D, #366. Long term fix is included in this [PR](https://github.com/processing/processing4/pull/367)

**Comment by benfry** [comment]
Fix incorporated for 4.0 beta 4.
